probe-rs/src/debug/function_die.rs

```diff
@@ -59,11 +58,29 @@ impl<'debugunit, 'abbrev, 'unit: 'debugunit, 'unit_info> FunctionDie<'abbrev, 'u
             unit_info,
             function_die: concrete_die,
             abstract_die: Some(abstract_die),
-            low_pc: 0,
-            high_pc: 0,
+            ranges: Vec::new(),
         })
     }
 
+    /// Test whether the given address is contained in the address ranges of this function.
+    /// Use this, instead of checking for values between `low_pc()` and `high_pc()`, because
+    /// the address ranges can be disjointed.
+    pub(crate) fn range_contains(&self, address: u64) -> bool {
+        self.ranges.iter().any(|range| range.contains(&address))
+    }
+
+    /// Returns the lowest valid address for which this function DIE is valid.
+    /// Please use `range_contains()` to check whether an address is contained in the range.
+    pub(crate) fn low_pc(&self) -> Option<u64> {
+        self.ranges.first().map(|range| range.start)
```

Comment:
Are these guaranteed to be sorted?

Reply:
Yep. DWARF spec 2.17 literally says "the first address in the first range entry in the list of ranges" :-)